client/db.go

```diff
@@ -290,14 +296,33 @@ type GQLResult struct {
 	//
 	// If there are values in this slice the request will likely not have run to completion
 	// and [Data] will be nil.
-	Errors []error `json:"errors,omitempty"`
+	Errors []GQLError `json:"errors,omitempty"`
 
 	// Data contains the resultant data produced by the GQL request.
 	//
 	// It will be nil if any errors were raised during execution.
 	Data any `json:"data"`
 }
```

Comment:
suggestion:  If you add a Marshal/Unmarshal func for `GQLResult` you could probably drop the new types and functions.  It might also be nicer for users of the Go client.

Reply:
I would just add a custom Marshal/Unmarshal func for `GQLError` instead .

Reply:
> I would just add a custom Marshal/Unmarshal func for GQLError instead .

I think the custom marshalling only works if its on the GQLResult and we want to keep errors as the generic `error` type.

I've refactored it to use a custom marshaller and added some tests to make sure our http server returns proper responses.